src/custom/hooks/useWrapCallback.ts

```diff
@@ -0,0 +1,112 @@
+import { TransactionResponse } from '@ethersproject/providers'
+import { getChainCurrencySymbols } from '@src/custom/utils/xdai/hack'
+import { Currency, CurrencyAmount, currencyEquals, ETHER, WETH } from '@uniswap/sdk'
+import { Contract } from 'ethers'
+import { useMemo } from 'react'
+import { tryParseAmount } from 'state/swap/hooks'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { useCurrencyBalance } from 'state/wallet/hooks'
+import { useActiveWeb3React } from 'hooks/index'
+import { useWETHContract } from 'hooks/useContract'
+
+export enum WrapType {
+  NOT_APPLICABLE,
+  WRAP,
+  UNWRAP
+}
+
+interface WrapUnwrapCallback {
+  wrapType: WrapType
+  execute?: undefined | (() => Promise<void>)
+  inputError?: string
+}
+
+type TransactionAdder = ReturnType<typeof useTransactionAdder>
+
+interface GetWrapUnwrapCallback {
+  isWrap: boolean
+  balance?: CurrencyAmount
+  inputAmount?: CurrencyAmount
+  addTransaction: TransactionAdder
+  wethContract: Contract
+}
+
+const NOT_APPLICABLE = { wrapType: WrapType.NOT_APPLICABLE }
+
+function _getWrapUnwrapCallback(params: GetWrapUnwrapCallback): WrapUnwrapCallback {
+  const { native, wrapped } = getChainCurrencySymbols()
+  const { isWrap, balance, inputAmount, addTransaction, wethContract } = params
+  const symbol = isWrap ? native : wrapped
+
+  // Check if user has enough balance for wrap/unwrap
+  const sufficientBalance = (inputAmount && balance && !balance.lessThan(inputAmount)) || false
+
+  // Create wrap/unwrap callback if sufficient balance
+  let wrapUnwrapCallback: (() => Promise<void>) | undefined
+  if (sufficientBalance && inputAmount) {
+    let wrapUnwrap: () => TransactionResponse
+    let summary: string, symbol: string
+
+    if (isWrap) {
+      wrapUnwrap = () => wethContract.deposit({ value: `0x${inputAmount.raw.toString(16)}` })
+      summary = `Wrap ${inputAmount.toSignificant(6)} ${native} to ${wrapped}`
+    } else {
+      wrapUnwrap = () => wethContract.withdraw(`0x${inputAmount.raw.toString(16)}`)
```

Comment:
concerned with magic number 16, i get it's probably the base number type but seems weird

Reply:
u know how much I dislike magic numbers, but in this case, I'm not sure we want to change it
at least we don't do it when we do `toString(10)`

I agree we can have RADIX_DECIMAL and RADIX_HEX for this numbers. But it feels strange we start now in just this PR to do it. Actually this PR is just factoring some parts of their hook, so I don't feel is too related.

Reply:
Happy to do the change if we think is best, but then we need to be consistent in other parts of the app and other apps.